react-app/components/DropZoneFile/index.jsx

```diff
@@ -19,43 +20,68 @@ const fileTypesAllowed = {
 const DropZoneFile = () => {
   const { setFilesToUploadList } = useContext(GalleryContext);
 
+  async function uploadAndFetchFiles(formData, _ = 10) {
+    try {
+      const { element: globalElement, xblockId } = globalObject;
+      const fileUploadHandler = globalObject.runtime.handlerUrl(globalElement, 'file_upload');
+
+      // Upload the file
+      const uploadResponse = await apiConfig.post(fileUploadHandler, formData, {
+        headers: {
+          'Content-Type': 'multipart/form-data'
+        }
+      });
+
+      if (uploadResponse.status === 200) {
+
+        const { data: imagesUploaded } = uploadResponse;
+
+        const formatImagesUploaded = imagesUploaded.map(({ id, asset_key, display_name, file_size, external_url }) => ({
+          id,
+          assetKey: asset_key,
+          name: display_name,
+          size: file_size,
+          url: external_url
+        }));
+
+        const filesSaved = getItemLocalStorage(xblockId) || [];
+
+        const filesUnloaded = [...filesSaved, ...formatImagesUploaded];
+        setFilesToUploadList(filesUnloaded);
+        setItemLocalStorage(xblockId, filesUnloaded);
+        // TODO: get data from backend when we have fixed the gallery data
+        /* // Fetch files
+         const data = {
+            current_page: 0,
+            page_size: pageSize
+          };
+
+        //const filesResponse = await apiConfig.post(fileGetterHandler, data);
+        // const {data: uploadedFiles } = filesResponse.files; */
+
+        // Handle the response here
+        // console.log(filesResponse.data);
+      } else {
+        console.error('File upload failed');
```

Comment:
As user, exist any kind of visual feedback about this error?

Reply:
Not yet, on it